src/components/library-app-2/book-form.js

```diff
@@ -1,218 +1,256 @@
-// import React, {useState} from 'react';
-// import CheckBox from '@react-native-community/checkbox';
-// import {
-//   StatusBar,
-//   StyleSheet,
-//   KeyboardAvoidingView,
-//   ScrollView,
-//   Pressable,
-//   Text,
-//   TextInput,
-//   View,
-// } from 'react-native';
-// import FontAwesome from 'react-native-vector-icons/FontAwesome';
-// import SelectDropdown from 'react-native-select-dropdown';
+import React, {useState} from 'react';
+import CheckBox from '@react-native-community/checkbox';
+import {
+  StatusBar,
+  StyleSheet,
+  KeyboardAvoidingView,
+  ScrollView,
+  Pressable,
+  Text,
+  TextInput,
+  View,
+  TouchableWithoutFeedback,
+  ActivityIndicator,
+  Platform,
+  Keyboard,
+  Alert,
+} from 'react-native';
+import FontAwesome from 'react-native-vector-icons/FontAwesome';
+import SelectDropdown from 'react-native-select-dropdown';
+import axios from 'axios';
 
-// const data = ['Vishal', 'nishant', 'Vicky'];
+const dropdownData = ['Vishal', 'Nishant', 'Vicky'];
+const url = 'https://library-mock-server.herokuapp.com/';
 
-// function BookForm() {
-//   const [formValues, setFormValues] = useState({
-//     bookName: '',
-//     authorName: '',
-//     publisher: '',
-//     price: '',
-//     email: '',
-//     website: '',
-//     displayBook: false,
-//   });
+const reqFields = ['bookName', 'authorName', 'price', 'email', 'website'];
```

Comment:
Required Fields, defaultFormValues, formFields ..... these should come from a single Form Object. Restructure your Object, think via the form library approach u did earlier

Reply:
First think ureself. Then lets discuss

Reply:
done